lib/msal-common/src/request/AuthorizationCodeRequest.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * @type AuthorizationCodeRequest: Request object passed by user to acquire a token from the server exchanging a valid authorization code
+ *  (second leg of OAuth2.0 Authorization Code flow)
+ *
+ * scopes:                  A space-separated array of scopes for the same resource.
+ *
+ * authority:               URL of the authority, the security token service (STS) from which MSAL will acquire tokens.
+ *
+ * redirectUri:             The redirect URI of your app, where the authority will redirect to after the user inputs credentials
+ *                          and consents. It must exactly match one of the redirect URIs you registered in the portal.
+ *
+ * code:                    The authorization_code that the user acquired in the first leg of the flow.
+ *
+ * codeVerifier:            The same code_verifier that was used to obtain the authorization_code.
+ *                          Required if PKCE was used in the authorization code grant request.
+ *                          For more information, see the PKCE RFC: https://tools.ietf.org/html/rfc7636
+ *
+ * correlationId:           Unique GUID set per request to trace a request end-to-end for telemetry purposes
+ *
+ */
+export type AuthorizationCodeRequest = {
```

Comment:
I would consider naming this `AuthorizationTokenRequest` instead of `AuthorizationCodeRequest`, I was a bit confused by this.

Reply:
AuthorizationCode is the name set by Oauth2.0. When we add the API that has both steps, it'll be something along the lines of InteractiveRequest. This is aligned with .Net, Python, and Java.